gateway/src/shard/processor/session.rs

```diff
@@ -137,14 +141,14 @@ impl Session {
         let commands_allotted = u32::from(available_commands_per_interval(new_heartbeat_interval));
 
         // We can ignore an error if the ratelimiter has already been set.
-        let _result = self.ratelimit.set(
+        let _result = self.ratelimit.set(Some(
```

Comment:
We should document the behavior of this method for ourselves when scanning code, this might be a source of a bug down the line otherwise.